### PR TITLE
Introduction to Grails Development with IntellJ

### PR DESCRIPTION
**Pull request by tambastar**
### Article Checklist
**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process, quality, and style-once published, you can then submit any number of articles.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by tambastar** [comment]
fixes #1320

**Comment by tambastar** [comment]
This fixes #1320

File: Introduction-to-Grails-Development-with-Intellij/index.md [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+
+- Download [Grail Framework](https://grails.org/download.html) as per the tutorial we were at version 4.0.6 in which you will get a zipped file.
+- Extract the zipped file and place it where you keep development tools. This is how to do Windows [Unzip](https://www.windowscentral.com/how-zip-and-unzip-files-windows-10). Ubuntu [unzip](https://askubuntu.com/questions/499807/how-to-unzip-tgz-file-using-the-terminal)
+- Set environment variable `GRAILS_HOME` to point where Grails installation directory is located. [Set windows environment variable](https://www.computerhope.com/issues/ch000549.htm).[ Set Ubuntu environment variable](https://askubuntu.com/questions/58814/how-do-i-add-environment-variables)
+
+### Getting Started
+
+Grails provides a set of commands to support application development.IDE has also easen the use of commands and provided a way users can click menus for fast application development.
+
+To create an application in Grail framework `create-app` command can be passed in terminal or windows CMD with arguments like the name of the application.
```
Comment (paulodhiambo) [inline Introduction-to-Grails-Development-with-Intellij/index.md:29]: Consider rewriting "... Grail framework `create-app` …" to "... Grails framework `create-app`..."

File: Introduction-to-Grails-Development-with-Intellij/index.md [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+├── init
+├── services
+├── taglib
+├── utils
+└── views
+```
+
+### assets folder
+
+This is where all static folders are kept like javascript, Images, CSS and other static multimedia content
```
Comment (paulodhiambo) [inline Introduction-to-Grails-Development-with-Intellij/index.md:101]: Consider rewriting "This is where all static folders are ..." to "This is where all static files are ..."

File: Introduction-to-Grails-Development-with-Intellij/index.md [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+    Integer position
+        static constraints = {
+    }
+}
+
+```
+
+### Creating Controller Class
+
+The below command adds an empty Controller Class
```
Comment (paulodhiambo) [inline Introduction-to-Grails-Development-with-Intellij/index.md:184]: Change "The below command ..." to "The command below ... "

File: Introduction-to-Grails-Development-with-Intellij/index.md [hunk 4]
```diff
@@ -0,0 +1,293 @@
+### Getting Started with Grails Java Framework
+
+In recent years, many developers are using frameworks when coming up with websites. Frameworks have assisted in making sure there is  faster application deployment and also coming up with quality code. Java supports multiplatform including a desktop application, web application, mobile app development, and many more. In the web application, it can be embedded with Javascript, HTML, CSS, and many front end applications.
```
Comment (paulodhiambo) [inline Introduction-to-Grails-Development-with-Intellij/index.md:3]: Consider rewriting "In recent years, many developers are using frameworks when coming up with websites." to "In recent years, many developers are using frameworks when developing websites."

File: Introduction-to-Grails-Development-with-Intellij/index.md [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+
+Grails is superb because it makes developers concentrate on application requirements and use less time in configuring the framework. Grails provides tools for development and it is build based on tools like  Quarts, Hibernate, Spring and uses Gradle for library management.
+
+There are many Grails framework version being built which has the support of different libraries and tools
+
+### Prerequisites
+
+- IntelliJ integrated development environment(IDE)
+- Knowledge in Java development
+- Java development kit(JDK) installed in your computer/laptop to help in application development
```
Comment (paulodhiambo) [inline Introduction-to-Grails-Development-with-Intellij/index.md:17]: Consider providing download links to JDK and IntelliJ. Specify the Java version i.e. Java 11.

**Review by paulodhiambo — CHANGES_REQUESTED** [review]
@tambastar amazing article on Grails framework. I have mentioned a few suggestions, review and ping me when done.

File: Introduction-to-Grails-Development-with-Intellij/index.md (hunk 1 above)
Comment (tambastar) [reply Introduction-to-Grails-Development-with-Intellij/index.md:29]: sorted

File: Introduction-to-Grails-Development-with-Intellij/index.md (hunk 2 above)
Comment (tambastar) [reply Introduction-to-Grails-Development-with-Intellij/index.md:101]: sorted

File: Introduction-to-Grails-Development-with-Intellij/index.md (hunk 3 above)
Comment (tambastar) [reply Introduction-to-Grails-Development-with-Intellij/index.md:184]: sorted

File: Introduction-to-Grails-Development-with-Intellij/index.md (hunk 5 above)
Comment (tambastar) [reply Introduction-to-Grails-Development-with-Intellij/index.md:17]: solved

**Comment by tambastar** [comment]
@paulodhiambo sorted

**Comment by paulodhiambo** [comment]
Change your articles folder name to  "introduction-to-grails-development-with-Intellij" from  "Introduction-to-Grails-Development-with-Intellij". Make sure you go through [getting started guide](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/UPLOAD_INSTRUCTIONS.md).

**Comment by paulodhiambo** [comment]
@tambastar amazing article on introduction to the Grails framework. Check out the changes I have made.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
@tambastar please take keen note of the changes made during my final review. 

I found the article a bit difficult to understand at times.

This may lead to more revisions (& tiered article payouts) and/or PR closure. 

cc @paulodhiambo  please do the same as this will help improve revision quality and what to look for.

**Comment by tambastar** [comment]
Noted @hectorkambow

**Pull request merged**